app/controllers/secrets_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SecretsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    secret = Secret.new(secret_params)
+    if current_user.id != secret_params[:receiver_id] && secret.save
```

Comment:
This "not to same person" check is repeated in a couple of places. Maybe put it on the model as a validation that sender != receiver?